feign-reactor-cloud/src/main/java/reactivefeign/cloud/publisher/RibbonPublisherClient.java

```diff
@@ -73,10 +74,18 @@ public RibbonPublisherClient(LoadBalancerCommandFactory loadBalancerCommandFacto
     }
 
     protected ReactiveHttpRequest loadBalanceRequest(ReactiveHttpRequest request, Server server) {
-        URI lbUrl = UriComponentsBuilder.fromUri(request.uri())
+        String scheme = "http";
+        if (server instanceof DiscoveryEnabledServer
+            && ((DiscoveryEnabledServer) server).getInstanceInfo() != null
+            && server.getPort() == ((DiscoveryEnabledServer) server).getInstanceInfo().getSecurePort())
+            scheme = "https";
+        URI lbUrl =
+            UriComponentsBuilder.fromUri(request.uri())
                 .host(server.getHost())
+                .scheme(scheme)
```

Comment:
can we simply use here `server.getScheme()` instead all this logic

Reply:
It seems Ribbon is also not populating that field... and would require a Pull Request to their project for this to work. :(

Reply:
Can you find out how open feign had fixed this issue?

Reply:
Sure, will look into it and reply here.

Reply:
It is actually a very similar solution in `DefaultServerIntrospector`:
```
public boolean isSecure(Server server) {
    return this.serverIntrospectorProperties.getSecurePorts().contains(server.getPort());
  }
```
called from `EuerekaServerIntrospector`:
```
public boolean isSecure(Server server) {
    if (server instanceof DiscoveryEnabledServer) {
      DiscoveryEnabledServer discoveryServer = (DiscoveryEnabledServer)server;
      return discoveryServer.getInstanceInfo().isPortEnabled(PortType.SECURE);
    } else {
      return super.isSecure(server);
    }
  }
```
called from `RibbonUtils`:
```
public static boolean isSecure(IClientConfig config, ServerIntrospector serverIntrospector, Server server) {
    if (config != null) {
      Boolean isSecure = (Boolean)config.get(CommonClientConfigKey.IsSecure);
      if (isSecure != null) {
        return isSecure;
      }
    }

    return serverIntrospector.isSecure(server);
  }
```
called from `RibbonUtils`:
```
  public static URI updateToSecureConnectionIfNeeded(URI uri, IClientConfig config, ServerIntrospector serverIntrospector, Server server) {
    String scheme = uri.getScheme();
    if (StringUtils.isEmpty(scheme)) {
      scheme = "http";
    }

    return !StringUtils.isEmpty(uri.toString()) && unsecureSchemeMapping.containsKey(scheme) && isSecure(config, serverIntrospector, server) ? upgradeConnection(uri, (String)unsecureSchemeMapping.get(scheme)) : uri;
  }
```
called from `FeignLoadBalancer`:
```
public URI reconstructURIWithServer(Server server, URI original) {
    URI uri = RibbonUtils.updateToSecureConnectionIfNeeded(original, this.clientConfig, this.serverIntrospector, server);
    return super.reconstructURIWithServer(server, uri);
  }
```

Reply:
I think it should be fixed on Spring side. 
Can you use "https" in URL as workaround?

Reply:
Well, the fix is on the reactive feign spring integration module(cloud module), just like spring-cloud-netflix-ribbon library is an integration library... so, to me, it is fitting. I didn't want to add additional dependencies neither, but the "cloud1" is specifically to be used with ribbon and, optionally, eureka.
I guess if we want to let Spring to fix it, then we would need to have `LoadBalancerCommand` to accept a `AbstractLoadBalancerAwareClient` as a way to reach the loadbalancer, and use spring implementation `FeignLoadBalancer` wrapper to the ILoadBalancer instead... as it is able to automatically upgrade to https when necessary.
But I think this fix will impact reactive-feign way more than what I proposed.

In another note, if I add "https" to the url it won't use the loadbalancer...

Reply:
Are there any updates on this?  I am in dire need of this capability.

Reply:
Me too..

Reply:
I suggest you do what I did. Compile and package your own version with the fix on this PR.